### PR TITLE
Reformat codebase

### PR DESCRIPTION
**Pull request by Scrumplex**

### CODE REVIEW

**Comment by Scrumplex** [comment]
Casually getting rid of 4k lines lol

**Comment by Scrumplex** [comment]
I thought about using this GH Action: https://github.com/DoozyX/clang-format-lint-action

**Comment by TheKodeToad** [comment]
I was thinking: if this causes compilation errors I have major concerns.
Then it happened

Anyway, maybe this is a good time to confirm that this is definitely what we want the formatting profile to be?

**Comment by Trial97** [comment]
I had a similar stuff when formating the codebase.
For me was just adding some missing headers: https://github.com/PrismLauncher/PrismLauncher/commit/1e18f2fc0f4c434c56e7a70033c57ecc4c32ec27

**Comment by TheKodeToad** [comment]
Seems some headers don't explicitly include

File: launcher/JavaCommon.cpp
Comment (TheKodeToad) [reply launcher/JavaCommon.cpp]: >:(

**Comment by Scrumplex** [comment]
Also made clang-format reformat remaining code (Java, ObjC and JSON files)

**Comment by TheKodeToad** [comment]
image

ew no thanks

Could this be changed? I was using Eclipse formatter before, and this actually hurts me :P.

**Comment by Scrumplex** [comment]
> ew no thanks
> 
> Could this be changed? I was using Eclipse formatter before, and this actually hurts me :P.

Probably. Maybe you can play with the `.clang-format` file yourself and check what you like?

See https://clang.llvm.org/docs/ClangFormatStyleOptions.html

You can just add another section like this:

```yaml
---
Language: Java
YourOption: foo
YourOtherOption: bar
```

File: launcher/minecraft/Library.cpp
Comment (TheKodeToad) [reply launcher/minecraft/Library.cpp:77]: wontfix? already has enough comments imo

Comment (Scrumplex) [reply launcher/minecraft/Library.cpp:77]: Too lazy lol

**Comment by TheKodeToad** [comment]
Please forgive my force-pushings. I assume it's OK if I push to your branch?

**Comment by txtsd** [comment]
I've been waiting for this for so long! It was one of the first things I wanted to do back when we forked from MultiMC.

**Review by TheKodeToad — APPROVED** [review]

**Review by getchoo — APPROVED** [review]

**Comment by Scrumplex** [comment]
*rubs hands*

I am gonna destroy these 50 PRs' whole careers.

**Comment by Scrumplex** [comment]
As requested by @TheKodeToad, I also replaced the remaining PolyMC license headers with Prism Launcher!

**Comment by Ryex** [comment]
...
I asked you to wait for *one* PR, *one* and it got the approval it needed
now I have this
image

**Comment by Scrumplex** [comment]
I needed a guinea pig PR to test my workflow for handling these conflicts :D

**Pull request merged**
